wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,377 @@
+from discord.ext import commands
+import discord
+from resources.utilities.embed import embed as em
+import psycopg2
+import datetime
+import pytz
+import re
+from typing import Union
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class Ban(commands.Cog):
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self.blacklist = []
```

Comment:
what's the purpose of this vs using the DB for keeping a list of the banned users?

Reply:
i dont think its a good idea to grab list of ids from db every time a user joins the guild. so this list is used to check for banned users joining. the db is used to keep the list across bot restarts and to have a ban history.

Reply:
that's not a bad point, but it assumes that 

1. we get sudden influx of users at any given time which can overload the DB. This would only happen on the first week of Spring, Summer and Fall term, right?

And one problem I see with it is how often do you flush the blacklist to the DB to ensure if that wall_e experiences a technical error, it can restore blacklist state?

Reply:
>that's not a bad point, but it assumes that we get sudden influx of users at any given time which can overload the DB. This would only happen on the first week of Spring, Summer and Fall term, right?

no, the server insights indicate that our weekly average of new users joining is 1500~ when i check and it said that was lower than previous week. a local list to check against when someone joins makes the most sense.

>And one problem I see with it is how often do you flush the blacklist to the DB to ensure if that wall_e experiences a technical error, it can restore blacklist state?

the blacklist is never flushed to the db.

here what happens when someone is banned:
 1. mod bans user
 2. user id is added to black list
 3. user name and id is added to `banned_users` table
 4. user name, id, mod name, mod id, datetime, reason for ban is added to `ban_records` table

so there is no need to flush blacklist to the db.

the blacklist is loaded from the db when the bot restarts in the load function.

Reply:
can you name it ban list then? just to synchronize it with the name for the DB table.